cg/services/post_processing/pacbio/metrics_parser/utils.py

```diff
@@ -32,47 +32,30 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-@handle_pac_bio_parsing_errors
-def parse_report_to_model(report_file: Path, data_model: Type[BaseMetrics]) -> BaseMetrics:
+def _get_data_model_from_pattern(pattern: str) -> Type[BaseMetrics]:
+    """Return the data model based on the pattern."""
+    pattern_to_model = {
+        PacBioDirsAndFiles.BASECALLING_REPORT: HiFiMetrics,
+        PacBioDirsAndFiles.CONTROL_REPORT: ControlMetrics,
+        PacBioDirsAndFiles.LOADING_REPORT: ProductivityMetrics,
+        PacBioDirsAndFiles.RAW_DATA_REPORT: PolymeraseMetrics,
+        PacBioDirsAndFiles.SMRTLINK_DATASETS_REPORT: SmrtlinkDatasetsMetrics,
+    }
+    return pattern_to_model.get(pattern)
+
+
+def _parse_report_to_model(report_file: Path, data_model: Type[BaseMetrics]) -> BaseMetrics:
     """Parse the metrics report to a data model."""
-    parsed_json: dict = ReadFile.read_file[FileFormat.JSON](file_path=report_file)
+    parsed_json: dict = ReadFile.read_file[FileFormat.JSON](report_file)
+    if data_model == SmrtlinkDatasetsMetrics:
```

Comment:
@ChrOertlin The only downside of this approach is that there is less flexibility when adding a new file to parse if the file has a different structure, but other than that it encapsulates a lot of logic

Reply:
I think it is also a bit more difficult to understand what is going on, but sure it removes some duplication 😄

Reply:
You can get the function and the model from a map via the file name. This should not need any IF statement

Reply:
@ChrOertlin  All the other reports have the data we want in a section of the JSON called `attributes`. Those JSON files have the data of the plots, so they have some other figure metadata This smrlink dataset is not a file containing data shown in any plot, it seems to hold run metadata, and it does not have this structure

Reply:
Lets leave this as it is and if we get rouble further down we can change it